FrontEnd/src/components/ProfilePage/FormComponents/AdditionalInfo.js

```diff
@@ -0,0 +1,114 @@
+import css from './FormComponents.module.css';
+import { useState, useEffect } from 'react';
+
+import HalfFormField from './FormFields/HalfFormField';
+import TextField from './FormFields/TextField';
+
+const LABELS = {
+    'foundationYear': 'Рік заснування',
+    'companySize': 'Розмір компанії',
+    'topClients': 'Топ клієнти',
+    'passedAudit': 'Пройдений аудит',
+};
+
+const AdditionalInfo = (props) => {
+    const [user, setUser] = useState(props.user);
+    const [foundationYearError, setFoundationYearError] = useState(null);
+    const maxLength = 1000;
+
+    useEffect(() => {
+        props.CurrentFormNameHandler('AdditionalInfo');
+    }, []); 
+
+    const onUpdateTextAreaField = e => {
+        if (e.target.value.length <= maxLength)
+            setUser((prevState) => {
+                return { ...prevState, [e.target.name]: e.target.value };
+            });
+    };
+
+    const onUpdateField = e => {
+        setUser((prevState) => {
+            return { ...prevState, [e.target.name]: e.target.value };
+        });
+    };
+
+    const onUpdateFoundationYearField = e => {
+        const currentYear = new Date().getFullYear();
+        const year = Number(e.target.value);
+        if ((1800 <= year && year <= currentYear) || !e.target.value) {
+            setFoundationYearError(null);
+        } else {
+            setFoundationYearError(`Рік заснування не в діапазоні 1800-${currentYear}`);
+        }
+        setUser((prevState) => {
+            return { ...prevState, [e.target.name]: e.target.value };
+        });
+    };
+
+    const ValidateForm = () => {
```

Comment:
method name should start from lower-case. Also [react-hook-form](https://www.npmjs.com/package/react-hook-form) could simplify it

Reply:
Thanks, fixed)